Sources/Plasma/FeatureLib/pfMetalPipeline/plMetalEnumerate.mm

```diff
@@ -76,12 +74,55 @@ void plMetalEnumerate::Enumerate(std::vector<hsG3DDeviceRecord>& records)
 
         devRec.SetLayersAtOnce(8);
 
-        // Just make a fake mode so the device selector will let it through
-        hsG3DDeviceMode devMode;
-        devMode.SetWidth(hsG3DDeviceSelector::kDefaultWidth);
-        devMode.SetHeight(hsG3DDeviceSelector::kDefaultHeight);
-        devMode.SetColorDepth(hsG3DDeviceSelector::kDefaultDepth);
-        devRec.GetModes().emplace_back(devMode);
+        plDisplayHelper* displayHelper = plDisplayHelper::CurrentDisplayHelper();
+        hsG3DDeviceMode defaultMode;
+        for (const auto& mode : displayHelper->GetSupportedDisplayModes(mainDisplay)) {
+            hsG3DDeviceMode devMode;
+            devMode.SetWidth(mode.Width);
+            devMode.SetHeight(mode.Height);
+            devMode.SetColorDepth(mode.ColorDepth);
+            devRec.GetModes().emplace_back(devMode);
+        }
+
+        // now inspect the GPU and figure out a good default resolution
+        // This code is in Metal (for now) - but it should also work
+        // under OpenGL/Mac as long as the GPU also supports Metal.
+        if (@available(macOS 10.15, *)) {
```

Comment:
We have a `HAVE_BUILTIN_AVAILABLE` macro that's set for when `@available` is defined (or `__builtin_available` in C++)

I doubt this is ever getting built on compilers without that, but it might be good to wrap it for consistency

Reply:
I'll go ahead and wrap it. If this were code out in some generic or OpenGL part of the client I'd be more worried. But this being Metal code keeps it in a certain compiler range.